cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/apply/RequestsPlaner.java

```diff
@@ -0,0 +1,94 @@
+package cloud.foundry.cli.logic.apply;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import cloud.foundry.cli.crosscutting.exceptions.ApplyException;
+import cloud.foundry.cli.logic.diff.change.container.CfContainerChange;
+import cloud.foundry.cli.logic.diff.change.map.CfMapChange;
+import cloud.foundry.cli.logic.diff.change.object.CfNewObject;
+import cloud.foundry.cli.logic.diff.change.object.CfObjectValueChanged;
+import cloud.foundry.cli.logic.diff.change.object.CfRemovedObject;
+import reactor.core.publisher.Mono;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+
+/**
+ * This is the super class of all request planer classes which are responsible to build the required requests
+ * for their operation domain according to the given CfChange objects.
+ * The class does create the request tasks by implementing the {@link CfChangeVisitor} interface.
+ */
+public abstract class RequestsPlaner implements CfChangeVisitor {
+
+    public static final String CHANGE_TYPE_IS_NOT_SUPPORTED = "Change type is not supported.";
+
+    private final List<Mono<Void>> requests;
+
+    protected List<Mono<Void>> getRequests() {
+        return Collections.unmodifiableList(this.requests);
+    }
+
+    protected RequestsPlaner() {
```

Comment:
I would put: 

- attributes 
- constructor 
- methods. 

You should put the constructor above the "getRequests()" method.